src/py/pyodide/_package_loader.py

```diff
@@ -149,6 +155,25 @@ def should_load_dynlib(path: str):
     return not PLATFORM_TAG_REGEX.match(tag)
 
 
+def set_wheel_installer(
+    filename: str,
+    archive: IO[bytes],
+    target_dir: Path,
+    installer: str | None,
+    source: str | None,
+):
+    z = ZipFile(archive)
+    name, version = filename.split("-", 2)[:-1]
+    dist_info_name = f"{name}-{version}.dist-info"
+    if not (zipfile.Path(z) / dist_info_name).is_dir():
+        raise Exception("archive is not a valid wheel")
+    dist_info = target_dir / dist_info_name
+    if installer:
+        (dist_info / "INSTALLER").write_text(installer)
```

Comment:
Could you add an inline comment linking to https://packaging.python.org/en/latest/specifications/recording-installed-packages/#the-dist-info-directory to indicate that this is not an arbitrary name.